### PR TITLE
Fix javadoc and build warnings

### LINKED ISSUE #472: Fix javadoc warnings & errors
*Why?*

there are some warnings when running `maven install`. these warnings are related to javadoc. it would be good we can clean up at some time.

### PR DESCRIPTION
**Pull request by sudoHackIn**
Fix #472 #2155

### CODE REVIEW

**Review by Shoothzj — APPROVED** [review]

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java [hunk 1]
```diff
@@ -1680,7 +1680,7 @@ public ServerConfiguration setMajorCompactionThreshold(double threshold) {
     }
 
     /**
-     * Get the maximum milliseconds to run major compaction. If <= 0 the
+     * Get the maximum milliseconds to run major compaction. If {@literal <=}0</code> the
```
Comment (nicoloboschi) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java:1683]:
```suggestion
     * Get the maximum milliseconds to run major compaction. If {@literal <=} 0 the
```

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by Shoothzj** [comment]
we need to update the checkstyle plugin version first.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java (hunk 1 above)
Comment (sudoHackIn) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java:1683]: oh, sorry, my bad :(

**Comment by sudoHackIn** [comment]
> we need to update the checkstyle plugin version first.

To actual latest? [3.1.2?](https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-checkstyle-plugin/3.1.2)

**Review by nicoloboschi — APPROVED** [review]

**Comment by Shoothzj** [comment]
@sudoHackIn Sorry, I though the former error is caused by checkstyle version.

**Comment by Shoothzj** [comment]
`OWASP Dependency Check` failure: both `google-http-client-gson` and `maven-settings` are not related to this PR

**Review by Shoothzj — APPROVED** [review]

**Pull request merged**